CONTRIBUTING.md

```diff
@@ -56,14 +36,129 @@ When creating your enhancement request, please:
+        git checkout -b BRANCH
+        git push --set-upstream origin BRANCH
+
+    </details>
+
+4. **Test your changes**
+
+    <details><summary>Expand details 👇</summary><br/>
+
+    Our continuous integration (CI) testing runs [a number of checks](https://github.com/allenai/allennlp/actions?query=workflow%3APR) for each pull request on [GitHub Actions](https://github.com/features/actions). You can run most of these tests locally, which is something you should do *before* opening a PR to help speed up the review process and make it easier for us.
+    
+    First, you should run [`black`](https://github.com/psf/black) to make sure you code is formatted consistently. Many IDEs support code formatters as plugins, so you may be able to setup black to run automatically everytime you save. [`black.vim`](https://github.com/psf/black/tree/master/plugin) will give you this functionality in Vim, for example. But `black` is also easy to run directly from the command line. Just run this from the root of your clone:
+    
+        black .
+
+    Our CI also uses [`flake8`](https://github.com/allenai/allennlp/tree/master/tests) to lint the code base and [`mypy`](http://mypy-lang.org/) for type-checking. You should run both of these next with
+
+        make lint
+
+    and
+
+        make typecheck
+
+    We also strive to maintain high test coverage, so most contributions should include additions to [the unit tests](https://github.com/allenai/allennlp/tree/master/tests). These tests are ran with [`pytest`](https://docs.pytest.org/en/latest/), which you can use to locally run any test modules that you've added or changed.
+
+    For example, if you've fixed a bug in `allennlp/nn/util.py`, you can run the tests specific to that module with
+    
+        pytest -v tests/nn/util_test.py
+    
+    Our CI will automatically check that test coverage stays above a certain threshold (around 90%). To check the coverage locally in this example, you could run
+    
+        pytest -v --cov allennlp.nn.util tests/nn/util_test.py
+
+    You should also make sure that the API documentation can build without errors. For that, just run
+
+        make build-docs
+
+    You can even serve and view the docs locally with
+    
+        make serve-docs
+
+    And finally, please update the [CHANGELOG](https://github.com/allenai/allennlp/blob/master/CHANGELOG.md) with notes on your contribution in the "Unreleased" section at the top.
+
+    After all of the above checks have passed, you can now open [a new GitHub pull request](https://github.com/allenai/allennlp/pulls).
+    Make sure you have a clear description of the problem and the solution, and include a link to relevant issues.
+
+    We look forward to reviewing your PR!
+
+    </details>
+
+### New models
+
+**Do you have a new state-of-the-art model?**
+
+We are always looking for new models to add to our collection.  If you have trained a model and would like to include it with 
```

Comment:
I'm not actually sure this should be our recommendation.  Having other people's models live in third party repositories is a solution we should encourage.

Reply:
Should we just remove this whole section then?

Reply:
I think this is a question that we should answer.  I'm just not sure what the answer should be, and I would lean towards the answer being, "we add your model to a list of third-party repos that we display prominently somewhere".  In certain cases, we can add them to our models repo, or offer to have them in our demo, or something, but that's not necessary.

Reply:
One of our longer-term goals post-1.0 (and post-performance) is to have a better story for third-party models. For now, if someone has a cool one, we should at least keep track of it somewhere, maybe in an issue, so we can take it into account when we have time to tackle this.

Reply:
How about we create an issue and link to it from `CONTRIBUTING.md` and maybe also the main `README.md`?